src/calculateAverage.js

```diff
@@ -37,7 +37,37 @@ function applyCalculateAverage() {
    * @param {Object} options - optional
    */
   [].__proto__.calculateAverage = function(options) {
-    // write code here
+    if (options === undefined) {
+      return this.reduce((a, b) => a + b) / this.length;
+    } else if (options === null
+      || typeof (options) === 'string'
+      || typeof (options.accumulator) === 'string'
+      || options.length > 1) {
+      return undefined;
+    } else {
+      let sum = 0;
+      let counter = 0;
+
+      for (let i = 0; i < this.length; i++) {
+        if (options.accumulator !== undefined
+          && typeof (options.accumulator) === 'function') {
+          sum += options.accumulator(this[i], i, this);
+          counter++;
+        }
+
+        if (options.propertyName !== undefined
+          && this[i][options.propertyName] !== undefined) {
+          if (typeof (this[i][options.propertyName]) !== 'number') {
+            return undefined;
+          } else {
+            sum += this[i][options.propertyName];
+            counter++;
+          }
+        }
+      }
+
+      return (counter > 0) ? sum / counter : undefined;
+    }
```

Comment:
последняя вложенность не нужна, она будет в людом случае по дефолту.
```suggestion
    } 
      let sum = 0;
      let counter = 0;

      for (let i = 0; i < this.length; i++) {
        if (options.accumulator !== undefined
          && typeof (options.accumulator) === 'function') {
          sum += options.accumulator(this[i], i, this);
          counter++;
        }

        if (options.propertyName !== undefined
          && this[i][options.propertyName] !== undefined) {
          if (typeof (this[i][options.propertyName]) !== 'number') {
            return undefined;
          } else {
            sum += this[i][options.propertyName];
            counter++;
          }
        }
        

      return (counter > 0) ? sum / counter : undefined;
    }
```